sdk/include/opentelemetry/sdk/logs/logger.h

```diff
@@ -41,18 +40,40 @@ class Logger final : public opentelemetry::logs::Logger
    * Initialize a new logger.
    * @param logger_provider The logger provider that owns this logger.
    */
-  explicit Logger(std::shared_ptr<LoggerProvider> logger_provider) noexcept;
+  explicit Logger(opentelemetry::nostd::string_view name,
+                  std::shared_ptr<LoggerProvider> logger_provider) noexcept;
 
   /**
    * Writes a log record into the processor.
-   * @param record The record to write into the processor.
-   */
-  void log(const opentelemetry::logs::LogRecord &record) noexcept override;
+   * @param severity the severity level of the log event.
+   * @param name the name of the log event.
+   * @param message the string message of the log (perhaps support std::fmt or fmt-lib format).
+   * @param resource the resources, stored as a 2D list of key/value pairs, that are associated
+   * with the log event.
+   * @param attributes the attributes, stored as a 2D list of key/value pairs, that are associated
+   * with the log event.
+   * @param trace_id the trace id associated with the log event.
+   * @param span_id the span id associate with the log event.
+   * @param trace_flags the trace flags associated with the log event.
+   * @param timestamp the timestamp the log record was created.
+   * @throws No exceptions under any circumstances.   */
+  void Log(opentelemetry::logs::Severity severity,
+           nostd::string_view name,
+           nostd::string_view body,
+           const common::KeyValueIterable &resource,
+           const common::KeyValueIterable &attributes,
+           trace::TraceId trace_id,
+           trace::SpanId span_id,
+           trace::TraceFlags trace_flags,
+           core::SystemTimestamp timestamp) noexcept override;
 
 private:
   // The logger provider of this Logger. Uses a weak_ptr to avoid cyclic dependancy issues the with
   // logger provider
   std::weak_ptr<LoggerProvider> logger_provider_;
+
+  // The name of this logger
+  opentelemetry::nostd::string_view logger_name_;
```

Comment:
Note: you may have to take ownership of this string. 

You need to at least do one of two things:

1. Document in the constructor that the instantiator of the Logger is responsible for keeping this string accessible for the lifetime of the logger.
2. Copy the String into memory you own and can control the lifecycle.